zerver/lib/actions.py

```diff
@@ -1663,6 +1666,21 @@ def fetch_stream_subscriber_emails(stream):
         notify_subscriptions_added(user_profile, sub_pairs, fetch_stream_subscriber_emails)
 
     for stream in streams:
+        # Send a notification to private stream when new user is added
+        if stream.invite_only:
+            new_users_names = [user.full_name for user in users
+                               if (user.id, stream.id) in new_streams]
+            if len(new_users_names) == 0:
+                pass
+            elif len(new_users_names) == 1:
+                content = ", ".join(new_users_names) + " has joined"
+                internal_send_message(settings.NOTIFICATION_BOT, "stream",
+                                      stream.name, "Users", content, sendtonotsubscribed=True)
+            else:
+                content = ", ".join(new_users_names) + " have joined"
+                internal_send_message(settings.NOTIFICATION_BOT, "stream",
+                                      stream.name, "Users", content, sendtonotsubscribed=True)
```

Comment:
Can we include which user made the change in the notification text?  I think that seems relevant.